internal/cdi/cdi.go

```diff
@@ -126,6 +129,10 @@ func New(infolib info.Interface, nvmllib nvml.Interface, devicelib device.Interf
 		return nil, fmt.Errorf("failed to create nvcdi library: %v", err)
 	}
 
+	if len(c.imexChannels) > 0 {
+		c.cdilibs["imex-channel"] = c.newImexChannelSpecGenerator()
+	}
+
```

Comment:
If / when we integrate this with the nvidia-container-toolkit, will this just become another case of the loop below, i.e.
```

	for _, mode := range additionalModes {
		lib, err := nvcdi.New(
			nvcdi.WithInfoLib(c.infolib),
			nvcdi.WithLogger(c.logger),
			nvcdi.WithNVIDIACDIHookPath(c.nvidiaCTKPath),
			nvcdi.WithDriverRoot(c.driverRoot),
			nvcdi.WithDevRoot(c.devRoot),
			nvcdi.WithVendor(c.vendor),
			nvcdi.WithMode(mode),
		)
		if err != nil {
			return nil, fmt.Errorf("failed to create nvcdi library: %v", err)
		}
		c.cdilibs[mode] = lib
	}
```

Reply:
Yes, basically. We may need to define how we're passing mode-specific options through the API -- such as which channel to select.